larpandora/LArPandoraEventBuilding/LArPandoraShower/Tools/ShowerTrajPointdEdx_tool.cc

```diff
@@ -54,6 +55,11 @@ namespace ShowerRecoTools{
       bool fUseMedian;        //Use the median value as the dEdx rather than the mean.
       bool fCutStartPosition; //Remove hits using MinDistCutOff from the vertex as well.
 
+      bool fT0Correct;        // Whether to look for a T0 associated to the PFP
+      bool fSCECorrectPitch;  // Whether to correct the "squeezing" of pitch, requires corrected input
+      bool fSCECorrectEField; // Whether to use the local electric field, from SpaceChargeService, in recombination calc.
+      bool fSCEInputCorrected; // Whether the input has already been corrected for spatial SCE distortions
```

Comment:
I very faintly recall that the Gnocchi module worked out whether the space points had been pre-corrected on the fly, rather than relying on a fcl switch.  Would it be possible to do something similar here?

I guess the main concern with FCL switches here is that, if there was a paradigm shift upstream and space points were/were not SCE corrected then someone would have to remember to manually propagate the change here.

Reply:
I think the Gnocci module also relies on fcl switches, although both also require the SpaceCharge service to be configured consistently with the fcl parameters. For example, [here](https://github.com/etyley/larpandora/blob/caeb29e75a33b6a0b09e6218aec5e512c2c4aeaa/larpandora/LArPandoraEventBuilding/LArPandoraShower/Algs/LArPandoraShowerAlg.cxx#L372-L375) will throw an exception if the correction function is called without the SpaceCharge service being configured correctly, which should help to enforce some discipline.  

I agree that this is not an ideal solution as it requires manual reconfiguration, but am not aware of a better solution to this.

Reply:
In practice the SCE service is probably almost always going to be configured correctly at the reconstruction stage, which is the only place where this code will be called.
Still, consistency is a good thing.  As above, it would be good to have a campaign to update all areas to use an updated scheme for figuring out when to un-correct (or have an alternative scheme where un-correcting isn't needed).
Thanks Ed.